docs/operational_guide/upgrading_m3.md

```diff
@@ -0,0 +1,101 @@
+# Upgrading M3
+
+## Overview
+
+This guide explains how to upgrade M3 from one version to another (e.g. from 0.14.0 to 0.15.0) - both on Kubernetes as well as on bare metal.
+This includes upgrading:
+
+- m3dbnode
+- m3coordinator
+- m3query
+- m3aggregator
+
+## m3dbnode
+
+### Graphs to monitor
+
+While upgrading M3DB nodes, it's important to monitor the status of bootstrapping the individual nodes. This can be monitored using the [M3DB Node Details](https://grafana.com/grafana/dashboards/8126) graph.
+Typically, the `Bootstrapped` graph under `Background Tasks` and the graphs within the `CPU and Memory Utilization` give a good understanding of how well bootstrapping is going.
+
+### Non-Kubernetes
```

Comment:
I'm trying to think of a better way to phrase this than "non-Kubernetes", since I think it makes it seem like M3 is more Kubernetes-focused than it is. Maybe "Per-Host Deployments" or "Host-Level Deployments"? Not totally sure

Reply:
Bare metal?

Reply:
Ehh that's usually associated with non-cloud / non-VM environments IMO lol

Reply:
I think non-kube is fine then. i think it gets the point across.